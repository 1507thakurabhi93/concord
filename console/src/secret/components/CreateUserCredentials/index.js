import React from 'react';
import { Field, reduxForm } from 'redux-form';

import { Form, Card, Button, Segment } from 'semantic-ui-react';

const CreateUserCredentials = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <Segment> 
      <h2>Create User Credentials</h2>
      <form onSubmit={handleSubmit}>
        
        <div>
          <label>Concord ID</label>
            <Field
              name="ConcordId"
              component={Form.Input}
              type="text"
              placeholder="Concord ID"
            />
        </div>

        <div>
          <label>Username</label>
            <Field
              name="username"
              component={Form.Input}
              type="text"
              placeholder="UserName"
            />
        </div>

        <div>
          <label>Password</label>
            <Field
              name="password"
              component={Form.Input}
              type="password"
              placeholder="Password"
            />
        </div>

        <div>
          <Button type="submit" disabled={pristine || submitting}>Submit</Button>
        </div>

      </form>
    </Segment>
  );
};

export default reduxForm({
  form: 'CreateUserCredentials',
})(CreateUserCredentials);
