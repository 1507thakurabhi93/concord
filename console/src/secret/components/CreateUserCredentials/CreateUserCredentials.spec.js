import React from 'react';
import { shallow, mount } from 'enzyme';
import CreateUserCredentials from './';

import { Form } from 'semantic-ui-react';

describe('CreateUserCredentials Component', () => {

    it('should render with no errors', () => {
        shallow(<CreateUserCredentials />);
    });

    it('should render a text input for the Secret Name', () => {
        const wrapper = shallow(<CreateUserCredentials />);
        expect( wrapper ).toMatchSnapshot();
    });
    
});