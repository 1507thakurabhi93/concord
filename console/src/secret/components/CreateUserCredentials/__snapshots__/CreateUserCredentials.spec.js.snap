// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserCredentials Component should render a text input for the Secret Name 1`] = `
<Connect(Form(CreateUserCredentials))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="CreateUserCredentials"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
